New K website

shd101wyy opened the pull request:
![Screenshot from 2020-08-13 22-25-01](https://user-images.githubusercontent.com/1908863/90146632-db848100-ddb3-11ea-8014-bd75dfa45370.png)

Demo: https://shd101wyy.github.io/k

Then Jenkins task is trying to build the static website in `./web/public_content` directory, then move everything inside `./web/public_content` into `./`.

bmmoore commented:
@radumereuta, could you look at the failure? The first error is this:
    Failed tests: 
    ParserTest.testListAmbiguity:575 The error:  expected:<NOKLABEL(amb(#token(K,"-1"),NOKLABEL(#token(K,"1"))))> but was:<#token(K,"-1")>

bmmoore commented:
I see I missed some files, which seem to fix this. I'm recreating this with a cleaner branch.

shd101wyy committed:
feat: Initial commit for new website

shd101wyy committed:
doc: Updated tutorial markdown files

shd101wyy committed:
refactor: Updated Dockerfile to install nodejs

shd101wyy committed:
refactor: Removed apt-get install npm

shd101wyy committed:
refactor: Removed unused index.md

ehildenb commented on web/pages/tutorials.md:
```diff
@@ -0,0 +1,112 @@
+
+Here you will learn how to define a very simple functional language in K and the basics of how to use the K tool. The language is a call-by-value variant of lambda calculus with builtins and mu, and its definition is based on substitution.
+
+- [04'07"]   [Lesson 1, LAMBDA: Syntax Modules and Basic K Commands](./tutorial/1_k/1_lambda/lesson_1/README.html)
```
Here

dwightguth commented:
The static content is in serious need of making sure the out of date stuff is removed; that was one of the primary reasons we wanted to update the website.

ehildenb commented:
@shd101wyy we have decided a few things at the meeting, the gist of it is "Only keep up-to-date information about the current state of K." What that means is:

-   Please remove all links to the `tool/run` try it online tool for now.
-   Put the links to the tutorial videos at the _bottom_ of the page, and put some text telling people they are out of date.
-   Please remove all pages with publications/people/workshops (eg: http://www.kframework.org/K11/).

That should be a good start for now.

shd101wyy committed:
fix: Fixed the link generation

shd101wyy committed:
doc: Updated the tutorial docs

shd101wyy committed:
fix: Removed all pages with publications/people/workshops

shd101wyy commented:
Hi @ehildenb I have updated the website. Please see the demo here: https://shd101wyy.github.io/k/index.html 
Please let me know if I need to make any further changes. Thank you!

ehildenb commented:
@shd101wyy for the moment, let's also remove the "Projects", "News", "Events", and "Funding" pages from the site, since they are all out of date.

Also, please provide a link (in the left pane) called "User Documentation", which links to the file `pending-documentation.md` (in the K repo root), and another link called "Builtins" which links to `k-distribution/include/builtin/index.html` (generated from `k-distribution/include/builtin/README.md`).

ehildenb commented on web/pages/downloads.md:
```diff
@@ -0,0 +1,9 @@
+
+### KVM v3.4
+
+The 3.4 version of the K Virtual Machine [http://www.kframework.org/imgs/releases/kvm-3.4.zip](http://www.kframework.org/imgs/releases/kvm-3.4.zip) provides an already set up and working version of the K Framework.
```
```suggestion
```

We don't want to distribute old versions of K anymore.

shd101wyy committed:
doc: Removed Events, Funding, News, and Projects pages

shd101wyy committed:
doc: Added User Documentation and Builtins

shd101wyy committed:
refactor: Put unification.k back

shd101wyy committed:
feat: Added syntax highlighting for K

shd101wyy committed:
Merge branch 'master' into gh-pages-bootstrap-style

shd101wyy commented:
Hi @ehildenb , I have removed the out of date pages and added the User Documentation and Builtins pages. I also supported the syntax highlighting for K. Please take a look at the demo website when you have time. Thanks a lot!

https://shd101wyy.github.io/k

A page that contains example of syntax highlighting for K: https://shd101wyy.github.io/k/k-distribution/include/kframework/builtin/domains/

ehildenb commented:
Wow, this looks really great! Thanks @shd101wyy . I'm approving, but I believe you need @dwightguth 's approval to merge it because he's code-owner on some files you touched.

My thoughts are that any more fixes can be made note of and cleaned up in a follow-up PR within the next few weeks.

shd101wyy committed:
fix: Fixed one Jenkinsfile gh-pages deployment bug

shd101wyy committed:
fix: Formatted links in build-html.js to better fit crawler

shd101wyy committed:
doc: Updated the homepage

shd101wyy committed:
doc: Updated the homepage

shd101wyy committed:
doc: Added link to Builtins under Learn K

shd101wyy committed:
Merge branch 'master' into gh-pages-bootstrap-style

nishantjr commented on k-distribution/include/kframework/builtin/README.md:
```diff
@@ -20,5 +20,5 @@ the path prefix).
 -   [substitution](substitution.md): Hooked implementation of capture-aware
     sustitution for K definitions.
--   [unification](unification.k): Hooked implementation of unification
+-   [unification](https://github.com/kframework/k/blob/master/k-distribution/include/kframework/builtin/unification.k): Hooked implementation of unification
```
This should be `.md`? Also any reason this is an absolute path?

ehildenb replied:
@nishantjr please open a new issue or a PR instead.
